### PR TITLE
[DigitalTwins] Adds Support for API version 2023-02-27-preview

### PR DESCRIPTION
**Pull request by sjiherzig**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/digitaltwins/Azure.DigitalTwins.Core/api/Azure.DigitalTwins.Core.netstandard2.0.cs [hunk 1]
```diff
@@ -111,13 +111,12 @@ public partial class DigitalTwinsClient
...
     public partial class DigitalTwinsClientOptions : Azure.Core.ClientOptions
     {
-        public DigitalTwinsClientOptions(Azure.DigitalTwins.Core.DigitalTwinsClientOptions.ServiceVersion version = Azure.DigitalTwins.Core.DigitalTwinsClientOptions.ServiceVersion.V2022_05_31) { }
+        public DigitalTwinsClientOptions(Azure.DigitalTwins.Core.DigitalTwinsClientOptions.ServiceVersion version = Azure.DigitalTwins.Core.DigitalTwinsClientOptions.ServiceVersion.V2023_02_27_Preview) { }
         public Azure.Core.Serialization.ObjectSerializer Serializer { get { throw null; } set { } }
         public Azure.DigitalTwins.Core.DigitalTwinsClientOptions.ServiceVersion Version { get { throw null; } }
         public enum ServiceVersion
         {
-            V2020_10_31 = 1,
-            V2022_05_31 = 2,
```
Comment (rachelsophia) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/api/Azure.DigitalTwins.Core.netstandard2.0.cs:120]: Why delete this as opposed to just adding the new version (which is what it seems like we did for V2022_05_31 at least)

File: sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClientOptions.cs [hunk 2]
```diff
@@ -63,8 +58,7 @@ internal string GetVersionString()
         {
             return Version switch
             {
-                ServiceVersion.V2020_10_31 => "2020-10-31",
-                ServiceVersion.V2022_05_31 => "2022-05-31",
```
Comment (rachelsophia) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClientOptions.cs:67]: Why delete these as opposed to just adding the new version (which is what it seems like we did for V2022_05_31 at least)

File: sdk/digitaltwins/Azure.DigitalTwins.Core/src/Customized/DigitalTwinsRestClient.cs [hunk 3]
```diff
@@ -19,6 +19,234 @@ internal partial class DigitalTwinsRestClient
     {
         private const string DateTimeOffsetFormat = "MM/dd/yy H:mm:ss zzz";
 
+        /// <summary>
```
Comment (rachelsophia) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/src/Customized/DigitalTwinsRestClient.cs:22]: A little confused why these need to be added, seems like its related to other parts of the SDK not bulk right?

Comment (rachelsophia) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/src/Customized/DigitalTwinsRestClient.cs:22]: In other words - still a little confused about what changed in bulk to cause this - or is this something we should / could do as a separate PR? would it need more reviewing than bulk?

Comment (sjiherzig) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/src/Customized/DigitalTwinsRestClient.cs:22]: I _think_ this was caused by the additionalProperties that were added, which are generic dictionaries. I.e., the response type really changed from a generic object to a dictionary. In our SDKs we previously already accounted for this case to some extent, but additional changes are necessary with the updated response model. I think in the solutions client we have something similar.

File: sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClientOptions.cs (hunk 2 above)
Comment (sjiherzig) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClientOptions.cs:67]: There was parity in terms of features between the two versions. If we only add it, then we would need to add additional logic to hide / avoid calling e.g., bulk with the previous API versions. This is the generally recommended pattern.

**Pull request closed without merging**